src/components/button/index.jsx

```diff
@@ -0,0 +1,249 @@
+                    transparent 1%,
+                    ${darken(0.2, `${theme.alto}`)} 1%
+                );
+                background-color: ${darken(0.2, `${theme.alto}`)};
+                border-color: ${darken(0.2, `${theme.alto}`)};
+            }
+
+            &.disabled--clickable:active {
+                background-color: ${darken(0.1, `${theme.alto}`)};
+                border-color: ${darken(0.2, `${theme.alto}`)};
+            }
+        }
+    }
+
+    &.btn--link {
+        background: transparent;
+        color: ${(props) => props.color};
+        min-width: 0;
+
+        &:hover,
+        &:focus,
+        &:active {
+            background: transparent;
+            box-shadow: none;
+            color: ${(props) => darken(0.2, props.color)};
+            transition: none;
+        }
+    }
+`;
+
+const Content = styled.span`
+    display: flex;
+    align-items: stretch;
+    justify-content: center;
+`;
+
+const Icon = styled.span`
+    display: flex;
+    align-items: center;
+    order: ${(props) => (props.iconOnRight ? '2' : '0')};
+    padding-right: ${(props) => (props.iconOnRight ? '0' : '5px')};
+    padding-left: ${(props) => (props.iconOnRight ? '5px' : '0')};
+`;
+Icon.displayName = 'Icon';
+
+const Text = styled.span`
+    display: flex;
+    align-items: center;
+    padding-left: ${(props) => (props.iconOnRight ? '5px' : '0')};
+	padding-right: ${(props) => (props.iconOnRight ? '0' : '5px')};
+`;
+Text.displayName = 'Text';
+
+export const Button = (props) => {
+    const handleClick = (event) => {
+        if (props.handleClick) {
+            event.preventDefault();
+            props.handleClick(event);
+        }
+        return false;
+    };
+
+    const renderTitleOrText = () => {
+        if (props.title) {
+            return props.title;
+        }
+        if (props.buttonText) {
+            return props.buttonText;
+        }
+        return null;
+    };
+
+    const renderIcon = () => {
+        if (props.icon) {
+            return <Icon iconOnRight={props.iconOnRight}>{props.icon}</Icon>;
+        }
+        return null;
+    };
+
+    const renderText = () => {
+        if (props.buttonText) {
+            return <Text iconOnRight={props.iconOnRight}>{props.buttonText}</Text>;
+        }
+        return null;
+    };
+
+    return (
+        <ColoredButton
+            color={props.color}
+            colorAlt={props.colorAlt}
+            iconOnRight={props.iconOnRight}
+            onClick={handleClick}
+            className={props.className}
+            title={renderTitleOrText()}
+            type={props.type}
+            aria-label={renderTitleOrText()}
+            disabled={props.disabled}
+            hidden={props.hidden}
+            minWidth={props.minWidth}
+        >
+            <Content>
+                {renderIcon()}
+                {renderText()}
+            </Content>
+        </ColoredButton>
+    );
+};
+
+// export class Button extends Component {
```

Comment:
Might be a WIP commit, but just noticed some commented out code - you won't want that in the submission :D

Reply:
lol thanks just switched it to functional will remove :)

Reply:
Should be removed!